WcaOnRails/app/controllers/persons_controller.rb

```diff
@@ -33,7 +33,7 @@ def show
     @ranks_average = @person.ranksAverage
     @medals = @person.medals
     @records = @person.records
-    @results = @person.results.includes(:competition, :event, :format, :round_type).order("Events.rank, Competitions.start_date DESC, RoundTypes.rank DESC")
+    @results = @person.results.includes(:competition, :event, :format, :round_type).order("Events.rank, Competitions.start_date DESC, RoundTypes.rank DESC, Competitions.id")
```

Comment:
It's hard for me to tell how this will work (we go on to do a bunch of `group_by`s in ruby code, but I think it would be safer/more correct to instead put `Competitions.id` right after `start_date`:

```ruby
order("Events.rank, Competitions.start_date DESC, Competitions.id, RoundTypes.rank DESC")
```

Reply:
Agreed!